src/awkward/operations/convert.py

```diff
@@ -2904,7 +2904,7 @@ def handle_arrow(obj):
                 return ak.operations.structure.concatenate(arrays, highlevel=False)
 
         elif (
-            isinstance(obj, Iterable)
+            isinstance(obj, Sequence)
```

Comment:
Why not Iterable? Is it because Iterables are not guaranteed to have a length, and `len(obj)` fails with an exception?

If that's the case, then a more appropriately narrow guard would be `isinstance(obj, Iterable) and isinstance(obj, Sized)`. In the [Python collections ABCs](https://docs.python.org/3/library/collections.abc.html),

   * Iterable just means that it has `__iter__` (the `for x in obj` loops will work)
   * Sized just means that it has a `__len__` (the `len(obj) > 0` check will not raise an exception)
   * Sequence means that it has a `__len__` and a `__getitem__`. The `__getitem__` is not required by any other checks in the predicate or any steps in the body.

So just replace `isinstance(obj, Sequence)` with `(isinstance(obj, Iterable) and isinstance(obj, Sized))`; the `Sized` symbol can be imported the same way that `Sequence` is.

Reply:
I think we had the same thought; the two classes is more exact to our needs, but I also wondered whether sequence adequately covered our case. I'll change it!